app/cdap/components/ConnectionList/index.tsx

```diff
@@ -198,76 +280,138 @@ export default function ConnectionList() {
       dataForTabs[0]?.data?.length > 0 ? (
-        <SelectDatasetWrapper>
-          {dataForTabs.map((each, index) => {
-            const connectionIdRequired =
-              each.data.length && each.data.filter((el) => el.connectionId);
-            if (connectionIdRequired.length) {
-              connectionId = connectionIdRequired[0].connectionId;
-            }
-            if (index === 0) {
-              headerContent = headerForLevelZero();
-            } else {
-              headerContent =
-                refs.current[index]?.offsetWidth < refs.current[index]?.scrollWidth ? (
-                  <CustomTooltip title={dataForTabs[index - 1].selectedTab} arrow>
-                    <Box className={classes.beforeSearchIconClickDisplay}>
-                      <Typography
-                        variant="body2"
-                        className={classes.headerLabel}
-                        ref={(element) => {
-                          refs.current[index] = element;
-                        }}
+        <Box className={classes.connectionsWithInfo}>
+          <SelectDatasetWrapper>
+            {filteredData &&
+              Array.isArray(filteredData) &&
+              filteredData?.map((eachFilteredData, index) => {
+                if (eachFilteredData.data.filter((el) => el.connectionId).length) {
+                  connectionId = eachFilteredData.data.filter((el) => el.connectionId)[0]
+                    .connectionId;
+                }
+                if (index === 0) {
+                  headerContent = headerForLevelZero();
+                } else {
+                  headerContent = (
+                    <>
+                      <Box
+                        className={
+                          eachFilteredData.isSearching
+                            ? classes.hideComponent
+                            : classes.beforeSearchIconClickDisplay
+                        }
+                      >
+                        {headersRefs.current[index]?.offsetWidth <
+                        headersRefs.current[index]?.scrollWidth ? (
+                          <CustomTooltip title={dataForTabs[index - 1].selectedTab} arrow>
+                            <Typography
+                              variant="body2"
+                              ref={(element) => {
+                                headersRefs.current[index] = element;
+                              }}
+                            >
+                              {filteredData[index - 1].selectedTab}
+                            </Typography>
+                          </CustomTooltip>
+                        ) : (
+                          <Typography
+                            variant="body2"
+                            ref={(element) => {
+                              headersRefs.current[index] = element;
+                            }}
+                          >
+                            {filteredData[index - 1].selectedTab}
```

Comment:
this component JSX is becoming too big and nonreadable
plz refactor to more readable and break into minor components wherever possible
update how the conditional renderings are done
thank you

Reply:
Done